vllm/entrypoints/openai/rpc/client.py

```diff
@@ -242,47 +356,63 @@ async def generate(
 
                 # Stream back the results from the RPC Server.
                 while not finished:
+
                     message = await socket.recv()
                     request_output = cloudpickle.loads(message)
 
                     if isinstance(request_output, Exception):
                         # On exception, check if the server is still healthy.
                         # Use this to set the sync `is_running` and `errored`
                         # properties.
-                        try:
-                            await self.check_health()
-                        except Exception:
-                            self._errored = True
+
```

Comment:
sorry I think I commented on the wrong line before, I meant to suggest adding the `finished = True` here rather than within the `except` block below.

Reply:
but don't we want `finished = False`, such that we call `abort`?

Reply:
+ then we only set `finished=True` in the `except` block such that we don't call self.abort() below